Is this an ok implementation of our test double?

diff --git a/04a-test_doubles/coffee-cart/test/cart.spec.ts b/04a-test_doubles/coffee-cart/test/cart.spec.ts
index 286533b..695394c 100644
--- a/04a-test_doubles/coffee-cart/test/cart.spec.ts
+++ b/04a-test_doubles/coffee-cart/test/cart.spec.ts
@@ -22,6 +22,7 @@ describe("buyCoffee", () => {
 	class FakeWorkingReader implements IReader {
 		public charges: number[] = [];
 		charge = (n: number) => {
+			if (n < 0) throw new Error("Invalid charge!");
 			this.charges.push(n);
 			return true;
 		};
